src/elements/walletConnect/WalletConnectButton.tsx

```diff
@@ -1,26 +1,26 @@
 import { UseWalletConnect } from './useWalletConnect';
 import { shortenString } from 'utils/pureFunctions';
-import { Image } from 'components/image/Image';
 import { ReactComponent as IconWallet } from 'assets/icons/wallet.svg';
+import useENS from 'hooks/useENS';
+import Davatar from '@davatar/react';
 
 export const WalletConnectButton = ({
   handleWalletButtonClick,
   account,
   selectedWallet,
 }: UseWalletConnect) => {
-  const buttonText = account ? shortenString(account) : 'Connect Wallet';
+  const { ensName } = useENS(account || '');
+  const buttonText = account
+    ? ensName || shortenString(account)
```

Comment:
Whats the possible max length of ens names? Lets limit it to something around 14 char

Reply:
ENS name limit is 255 but I'll add truncate function to it 👍🏻

Reply:
<img width="221" alt="Screenshot 2021-12-22 at 4 18 30 PM" src="https://user-images.githubusercontent.com/69431456/147081053-06a75049-a3ec-4d99-b889-1bda46fb8fb3.png">

@pingustar WDYT?